src/metabase/util/malli/schema.clj

```diff
@@ -78,7 +72,19 @@
 
 (def NonBlankString
   "Schema for a string that cannot be blank."
-  (mu/with-api-error-message ::lib.schema.common/non-blank-string (deferred-tru "value must be a non-blank string.")))
+  (mu/with-api-error-message
+   ;; this is directly copied from [[:metabase.lib.schema.common/non-blank-string]] -- unfortunately using it here would
+   ;; mean we need a dependency of `util` on `lib` -- not worth it to save ~6 duplicate LoC. At some point in the future
+   ;; maybe we can get everyone to use one or the other or better yet make more specific schemas that describe their
+   ;; purpose like `:metabase.warehouses.schema/database-description`. Who knows?
+   [:and
```

Comment:
I don't really love duplicating code but if duplicating 5 LoC eliminates a circular dependency between two modules then it's worth it I think.

Reply:
Why cannot `lib` dependend on `util`? Some dependencies here (e.g., `t2`) feel "heavy" for `util`.